enhancements/microshift/gateway-api-support.md

```diff
@@ -0,0 +1,434 @@
+and is released as an optional operator in the Red Hat operators catalog for
+OLM. This operator deploys all of Istio's manifests and is capable of handling
+Istio control plane deployments, Istio CNIs and other components. Istio has
+first class support for Gateway API resources and is already supported by Red
+Hat.
+
+### Workflow Description
+**User** is a human user responsible for setting up and managing Edge Devices.
+**Application** is user's workload that intends to use routing capabilities.
+
+#### Installation and usage on RHEL For Edge (ostree)
+> In this workflow, it doesn't matter if the device is already running R4E with existing MicroShift cluster.
+> Deployment of new commit requires reboot which will force recreation of the Pod networking after adding Multus.
+
+1. User gathers all information about the networking environment of the edge device.
+1. User prepares ostree commit that contains:
+   - MicroShift RPMs.
+   - Gateway API for MicroShift RPM.
+   - Gateway CRs
+   - Application using mentioned Gateway CRs.
+1. User deploys the ostree commit onto the edge device.
+1. Edge device boots:
+1. (optional) Init procedures are configuring OS and networks.
+1. MicroShift starts.
+1. MicroShift applies Gateway API manifests.
+1. Istio controller becomes available.
+1. MicroShift applies Application's manifests that may include Gateways and routes.
+1. Gateway pods are created and OSSM inspects routes to accept them into each Gateway.
+1. Application's pods are running, they are reachable through the routes in the Gateway.
+
+#### Installation and usage on RHEL (rpm)
+##### Adding to existing MicroShift cluster (or before first start)
+1. MicroShift already runs on the device. (This step may not be true if its before first start)
+1. User installs `microshift-gateway-api` RPM
+1. User restarts MicroShift service.
+1. MicroShift starts and deploys all Gateway API resources from `manifests.d`.
+1. Istio controller becomes available.
+1. User creates Gateway and routes (HTTP or GRPC) CRs
+1. When Gateway pods are created, routes are picked up and configured by the controller which is already running.
+1. Application's pods are running, they are reachable through the routes in the Gateway.
+
+### API Extensions
+Most of the manifests are CRDs, and users are left only with creating Gateway,
+HTTPRoute and GRPCRoute resources. These are explained in the use cases in the
+[documentation](https://gateway-api.sigs.k8s.io/concepts/use-cases/#basic-northsouth-use-case).
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+Not applicable
+
+#### Standalone Clusters
+Not applicable
+
+#### Single-node Deployments or MicroShift
+Enhancement is solely intended for MicroShift.
+
+### Implementation Details/Notes/Constraints [optional]
+N/A
+
+#### Manifests
+Gateway API is just a set a CustomResourceDefinitions. In order to support it
```

Comment:
Well, there is quite a bit of work you skip over there.  Gateway API implementations must abide by the API contract (CRDS, CEL, and standards) and are said to be conformant when they pass and publish the Gateway API conformance tests.

Reply:
As this section is talking about manifests only I rephrased it to make the intent more clear.